watchtower/wtmock/client_db.go

```diff
@@ -213,16 +215,27 @@ func (m *ClientDB) ListClientSessions(tower *wtdb.TowerID,
 // optional tower ID can be used to filter out any client sessions in the
 // response that do not correspond to this tower.
 func (m *ClientDB) listClientSessions(tower *wtdb.TowerID,
-	_ ...wtdb.ClientSessionListOption) (
+	opts ...wtdb.ClientSessionListOption) (
 	map[wtdb.SessionID]*wtdb.ClientSession, error) {
 
+	cfg := wtdb.NewClientSessionCfg()
+	for _, o := range opts {
+		o(cfg)
+	}
+
 	sessions := make(map[wtdb.SessionID]*wtdb.ClientSession)
 	for _, session := range m.activeSessions {
 		session := session
 		if tower != nil && *tower != session.TowerID {
 			continue
 		}
 		sessions[session.ID] = &session
+
+		if cfg.PerAckedUpdate != nil {
+			for seq, id := range m.ackedUpdates[session.ID] {
```

Comment:
Should some of these fields be guarded by a `lock`?

Is it possible that while iterating `m.ackedUpdates` here it gets modified by `AckUpdate` or more sessions get added/deleted from the slice?

Reply:
`listClientSessions` is only ever called by exported functions that are already holding the lock :)

Reply:
I have added a note about this to the function comment now so that it is more clear :)